packages/core-mobile/app/new/features/bridge/screens/BridgeScreen.tsx

```diff
@@ -0,0 +1,698 @@
+        <TokenInputWidget
+          amount={receiveAmount}
+          balance={selectedAssetInTargetNetwork?.balance}
+          title="You receive"
+          token={selectedAssetInTargetNetwork}
+          network={targetNetwork}
+          formatInCurrency={formatInCurrency}
+          onAmountChange={handleAmountChange}
+          editable={false}
+        />
+      </Animated.View>
+    )
+  }, [
+    colors,
+    formatInCurrency,
+    handleAmountChange,
+    handleSelectTargetNetwork,
+    selectedAssetInTargetNetwork,
+    targetNetwork,
+    targetNetworks,
+    hasValidAmount,
+    estimatedReceiveAmount,
+    selectedBridgeAsset
+  ])
+
+  useEffect(() => {
+    if (
+      initialSourceNetworkChainId &&
+      initialTokenSymbol &&
+      !selectedBridgeAsset
+    ) {
+      const initialSourceNetwork = sourceNetworks.find(
+        network => network.chainId === Number(initialSourceNetworkChainId)
+      )
+      if (initialSourceNetwork) {
+        setSourceNetwork(initialSourceNetwork)
+        const token = assetsWithBalances?.find(
+          tk => (tk.symbolOnNetwork ?? tk.symbol) === initialTokenSymbol
+        )
+        if (token) {
+          handleSelectBridgeAsset(token)
+        }
+      }
+    }
+  }, [
+    initialSourceNetworkChainId,
+    initialTokenSymbol,
+    assetsWithBalances,
+    handleSelectBridgeAsset,
+    selectedBridgeAsset,
+    setSourceNetwork,
+    sourceNetworks
+  ])
+
+  useEffect(() => {
+    if (error) {
+      setBridgeError(error.message)
+    }
+  }, [error])
+
+  useEffect(() => {
+    const checkGaslessEligibility = async (): Promise<void> => {
+      if (!sourceNetwork?.chainId || !selectedBridgeAsset) {
+        setIsGaslessEligible(false)
+        return
+      }
+
+      try {
+        const isEligible = await GaslessService.isEligibleForChain(
+          sourceNetwork.chainId.toString()
+        )
+        setIsGaslessEligible(isEligible)
+      } catch (err) {
+        Logger.error(`[Bridge.tsx][checkGaslessEligibility]${err}`)
+        setIsGaslessEligible(false)
+      }
+    }
+
+    checkGaslessEligibility()
+  }, [sourceNetwork?.chainId, selectedBridgeAsset])
+
+  useEffect(() => {
+    if (!sourceNetwork) return
+
+    // Reset because a denomination change will change its value
+    setInputAmount(undefined)
+  }, [sourceNetwork, setInputAmount])
+
+  useEffect(() => {
+    if (selectedSourceNetwork) {
+      setSourceNetwork(selectedSourceNetwork)
+    }
+  }, [selectedSourceNetwork, setSourceNetwork])
+
+  useEffect(() => {
+    if (selectedTargetNetwork) {
+      setTargetNetwork(selectedTargetNetwork)
+    }
+  }, [selectedTargetNetwork, setTargetNetwork])
+
+  useEffect(() => {
+    return () => {
+      setSelectedSourceNetwork(undefined)
+      setSelectedTargetNetwork(undefined)
+      setSelectedAsset(undefined)
+    }
+  }, [setSelectedSourceNetwork, setSelectedTargetNetwork, setSelectedAsset])
+
+  useEffect(() => {
+    if (selectedAsset) {
```

Comment:
there are lots of useEffect here. could you add some comments to each/some of them? and if you could also extract them to a custom hook.